File: 2025-09-25 - ListaPrioridade/Exemplo_C#/Node.cs
// ------------------------------------------------------------------------
// Node.cs
// Classe que representa um nó da lista de prioridade
// ------------------------------------------------------------------------
public class Node
{
    // ---------------------------------------------------------------------
    // Atributos privados (usando _ para diferenciar dos públicos)
    // ---------------------------------------------------------------------
    private string _codigoDisciplina;
    private string _nomeDisciplina;
    private int _cargaHoraria;
    private int _semestre;
    private Node _proximo; // referência para o próximo nó

    // ---------------------------------------------------------------------
    // Construtor
    // ---------------------------------------------------------------------
    public Node(string codigoDisciplina, string nomeDisciplina, int cargaHoraria, int semestre)
    {
        this._codigoDisciplina = codigoDisciplina;
        this._nomeDisciplina = nomeDisciplina;
        this._cargaHoraria = cargaHoraria;
        this._semestre = semestre;
        this._proximo = null;
    }

    // ---------------------------------------------------------------------
    // Propriedades Públicas em camelCase
    // ---------------------------------------------------------------------
    public string codigoDisciplina
    {
        get { return _codigoDisciplina; }
        set { _codigoDisciplina = value; }
    }

    public string nomeDisciplina
    {
        get { return _nomeDisciplina; }
        set { _nomeDisciplina = value; }
    }

    public int cargaHoraria
    {
        get { return _cargaHoraria; }
        set { _cargaHoraria = value; }
    }

    public int semestre
    {
        get { return _semestre; }
        set { _semestre = value; }
    }

    public Node proximo
    {
        get { return _proximo; }
        set { _proximo = value; }
    }
}

File: 2025-10-02 - Dicionarios-Tabela Hash/Exemplo_Python/Node.py
# ------------------------------------------------------------------------
# Node.py
# Classe Node (No) que representa um elemento na tabela hash (encadeamento)
# Campos privados: codigoDisciplina, nomeDisciplina, cargaHoraria, periodoCurso
# Também possui referência privada para o próximo nó (usado no encadeamento)
# ------------------------------------------------------------------------

class Node:
   """
   Representa um nó contendo informações da disciplina.
   Usado nas listas encadeadas que compõem cada bucket da tabela hash.
   """

   def __init__(self, codigoDisciplina: str, nomeDisciplina: str, cargaHoraria: int, periodoCurso: int):
      # Atributos privados (name mangling)
      self.__codigoDisciplina = codigoDisciplina
      self.__nomeDisciplina = nomeDisciplina
      self.__cargaHoraria = cargaHoraria
      self.__periodoCurso = periodoCurso
      # Referência para encadeamento (próximo nó)
      self.__next = None

   # ---------------------------------------------------------------------
   # Getters e setters (camelCase)
   # ---------------------------------------------------------------------
   def getCodigoDisciplina(self) -> str:
      return self.__codigoDisciplina

   def setCodigoDisciplina(self, codigoDisciplina: str):
      self.__codigoDisciplina = codigoDisciplina

   def getNomeDisciplina(self) -> str:
      return self.__nomeDisciplina

   def setNomeDisciplina(self, nomeDisciplina: str):
      self.__nomeDisciplina = nomeDisciplina

   def getCargaHoraria(self) -> int:
      return self.__cargaHoraria

   def setCargaHoraria(self, cargaHoraria: int):
      self.__cargaHoraria = cargaHoraria

   def getPeriodoCurso(self) -> int:
      return self.__periodoCurso

   def setPeriodoCurso(self, periodoCurso: int):
      self.__periodoCurso = periodoCurso

   def getNext(self):
      return self.__next

   def setNext(self, nextNode):
      self.__next = nextNode

   # Representação para debug/impressão
   def __repr__(self):
      return f"Node({self.__codigoDisciplina}, {self.__nomeDisciplina}, {self.__cargaHoraria}h, Período {self.__periodoCurso})"
